### PR TITLE
Add profile config api

### LINKED ISSUE #11: [3pts] Create Profile Settings Api path
- POST request to update a logged-in user infos

### PR DESCRIPTION
**Pull request by aimalamiri**
## Add update profile API

PATCH `/profile/update`

`response`

```json
{
    "updated": {
        "id": 18,
        "first_name": "Aimal",
        "last_name": "Amiri",
        "role": "admin",
        "email": "meaimal@hotmail.com",
        "password_digest": "$2a$12$HzGUaR6OZKSvomMUawSmluKSHIEaSKMtdzdg/TQZiMOlaWXglR.3u",
        "created_at": "2022-09-19T15:05:26.060Z",
        "updated_at": "2022-09-21T15:00:29.492Z"
    }
}

```

closes #11

### CODE REVIEW

**Review by ichala — CHANGES_REQUESTED** [review]
🛑 REQUIRED CHANGES 🛑


@aimalamiri great job ,
- [x] there's an error should be handled which is **_if the new provided email exist already_** 
- [x]  you need to consider handling the error of server error wich is 

```Ruby
if user.update(profile_params)
render json: { updated: current_user } 
else
render json: { error: current_user } , status: :not_implemented
end
```

**Comment by aimalamiri** [comment]
> 🛑 REQUIRED CHANGES 🛑
> 
> @aimalamiri great job ,
> 
> * [x]  there's an error should be handled which is **_if the new provided email exist already_**
> * [x]   you need to consider handling the error of server error wich is
> 
> ```ruby
> if user.update(profile_params)
> render json: { updated: current_user } 
> else
> render json: { error: current_user } , status: :not_implemented
> end
> ```

Thanks for the tip. It's done.

**Comment by ichala** [comment]
✅ Approved for me ✅

**Comment by ichala** [comment]
✅ APPROVED FOR ME ✅

File: app/controllers/profile_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
+
+  private
+
+  def profile_params
+    params.permit(:first_name, :last_name, :email, :password)
+  end
+end
```
Comment (d2ndjim) [inline app/controllers/profile_controller.rb:19]: Clean and Organized implementation 👍

**Review by d2ndjim — APPROVED** [review]
@aimalamiri 
### APPROVED ✔️ ✔️ ✔️

File: app/controllers/profile_controller.rb (hunk 1 above)
Comment (aimalamiri) [reply app/controllers/profile_controller.rb:19]: Thank you 🙏

**Review by EidHachem — APPROVED** [review]
Great job @aimalamiri 

# Approved ✔️ 

It's time to merge 🎉

**Review by ichala — APPROVED** [review]

**Pull request merged**